### ISSUE TITLE
Container dependencies

### ISSUE THREAD

**Issue by M1cha** [lxc/lxd#10947]
Being able to specify dependencies between containers would be a very useful feature in LXD. Before a container can start, all of it's dependencies must have started and sent the READY-event.
We might also want to add support for "light" dependencies which only affect autostart, but won't prevent a container from starting, similar to systemd's `After` option in the service definition.

We already have:
- `boot.autostart` to force startup on boot
- `boot.autostart.priority` to specify startup order
- `boot.autostart.delay` to make containers "wait" longer until others are up

In case you want to specify dependencies, the last two are kind of a workaround and would not be required if we could specify dependencies by name directly. 
This could look like this:
```
boot.autostart.dependenies:
  - container0
  - container1
```

[Original post on discourse](https://discuss.linuxcontainers.org/t/container-dependencies/15193)

**Comment by M1cha** [comment]
If this is purely a question of priorities but you're otherwise fine with adding that feature, I could also try to implement it. It can't hurt to get more familiar with LXDs code :grin:

**Comment by tomponline** [comment]
If @stgraber is open to the idea then would be happy to assign to you.

**Comment by tomponline** [comment]
Ive assigned it to you @M1cha as @stgraber added it to later tag, suggesting he approves of the idea. 😀

**Comment by stgraber** [comment]
Yeah, I'm not generally opposed to it. We just need to be somewhat careful with handling of projects and in avoiding loops.

**Comment by nalcabio-tom** [comment]
Curious if there was ever any work done on this here or in incus? I assume this would be valid for a single-machine deployment, right?

**Comment by tomponline** [comment]
> Curious if there was ever any work done on this in lxd or incus? I assume this would be valid only for a single-machine deployment, right?

No work has been done on it at this time. You raise a good question about cluster behaviour.
